crates/worker/src/partition_processor_manager.rs

```diff
@@ -242,14 +243,16 @@ impl PartitionProcessorHandle {
     }
 }
 
+type ChannelStatusReaderList = Vec<(RangeInclusive<PartitionKey>, ChannelStatusReader)>;
+
 #[derive(Debug, Clone, Default)]
 pub struct MultiplexedInvokerStatusReader {
-    readers: Vec<(RangeInclusive<PartitionKey>, ChannelStatusReader)>,
+    readers: Arc<Mutex<ChannelStatusReaderList>>,
 }
 
 impl MultiplexedInvokerStatusReader {
     fn push(&mut self, key_range: RangeInclusive<PartitionKey>, reader: ChannelStatusReader) {
-        self.readers.push((key_range, reader));
+        self.readers.lock().unwrap().push((key_range, reader));
```

Comment:
I was wondering if a a RwLock would fit better here since I noticed this push happens only during starting of a partition processor which not frequent.

Reply:
I think you are right. Will change it.